frostsnap_core/src/coordinator.rs

```diff
@@ -140,25 +186,27 @@ impl FrostCoordinator {
     }
 
     pub fn mutate(&mut self, mutation: Mutation) {
-        event!(Level::DEBUG, gist = mutation.gist(), "mutating");
         if let Some(reduced_mutation) = self.apply_mutation(&mutation) {
+            event!(Level::DEBUG, gist = mutation.gist(), "mutating");
             self.mutations.push_back(reduced_mutation);
+        } else {
+            event!(Level::DEBUG, gist = mutation.gist(), "ignoring mutation");
         }
     }
 
     pub fn apply_mutation(&mut self, mutation: &Mutation) -> Option<Mutation> {
         fn ensure_key<'a>(
```

Comment:
Note: this never updates a key, so if we try restore the same key under a different purpose, it will not be stored. Probably desired behaviour